packages/markup-frontstage/src/services/SavedView.ts

```diff
@@ -147,22 +219,125 @@ export const createSavedViewData = (vp: Viewport): SavedViewData => {
+ */
+export const isSpatialSavedView = (view: SavedViewData) => {
+  return (view.is2d === undefined || !view.is2d) && "models" in view;
+};
+
+/**
+ * Determines whether or not given view is Drawing.
+ * @param view name fo the Saved View.
+ */
+export const isDrawingSavedView = (view: SavedViewData) => {
+  return view.is2d !== undefined && view.is2d && !("sheetProps" in view);
+};
+
+/**
+ * Determines whether or not given view is Sheet.
+ * @param view name of the Saved View.
+ */
+export const isSheetSavedView = (view: SavedViewData) => {
+  return view.is2d !== undefined && view.is2d && "sheetProps" in view;
+};
+
+/**
+ * Creates a drawing view state from the data object.
+ * @param iModelConnection the name of the IModel Connection.
+ * @param savedView the naem of the Saved View.
+ */
+export const createDrawingViewState = async (
+  iModelConnection: IModelConnection,
+  savedView: SavedViewData
+): Promise<DrawingViewState | undefined> => {
+  const props: ViewStateProps = {
+    viewDefinitionProps: savedView.viewDefinitionProps
+      ? JSON.parse(savedView.viewDefinitionProps)
+      : undefined,
+    categorySelectorProps: savedView.categorySelectorProps
+      ? JSON.parse(savedView.categorySelectorProps)
+      : undefined,
+    displayStyleProps: savedView.displayStyleProps
+      ? JSON.parse(savedView.displayStyleProps)
+      : undefined,
+  };
+  const viewState = DrawingViewState.createFromProps(
+    props,
+    iModelConnection
+  ) as DrawingViewState;
+  await viewState.load();
+  return viewState;
+};
+
+/**
+ * Creates a sheet view state from the data object.
+ * @param iModelConnection the name of the IModelConnection.
+ * @param savedView the naem of the Saved View.
+ */
+export const createSheetViewState = async (
+  iModelConnection: IModelConnection,
+  savedView: SavedViewData
+): Promise<SheetViewState | undefined> => {
+  if (
+    savedView.sheetProps === undefined ||
+    savedView.sheetAttachments === undefined
+  ) {
+    return undefined;
+  }
+  const props: ViewStateProps = {
+    viewDefinitionProps: savedView.viewDefinitionProps
+      ? JSON.parse(savedView.viewDefinitionProps)
+      : undefined,
+    categorySelectorProps: savedView.categorySelectorProps
+      ? JSON.parse(savedView.categorySelectorProps)
+      : undefined,
+    displayStyleProps: savedView.displayStyleProps
+      ? JSON.parse(savedView.displayStyleProps)
+      : undefined,
+    sheetProps: savedView.sheetProps
+      ? JSON.parse(savedView.sheetProps)
+      : undefined,
+    sheetAttachments: savedView.sheetAttachments,
+  };
+  const viewState = SheetViewState.createFromProps(props, iModelConnection);
+  await viewState.load();
+  return viewState;
+};
+
+/**
+ * Creates a markup saved view data from the viewport, it could return a
+ * View2d(Sheet or Drawings) or a View3d(Spatial views) object
+ * @param vp the name of the view port.
+ */
+export const createMarkupSavedViewData = (vp: Viewport): SavedViewData => {
```

Comment:
Why do you need this special markup one at all? Isn't it the same as createSavedView?

Reply:
CreateSavedViewData returns spatial viewState, I was thinking of using createSavedViewData to return (spatial, sheet and drawing) it'll break consumer's since that function is accessible. Hence I named createMarkupSavedViewData instead of CreateSavedViewData, please let me know if you have any good name suggestion?